src/JustSaying/Fluent/QueuePublicationBuilder`1.cs

```diff
@@ -67,9 +71,47 @@ public QueuePublicationBuilder<T> WithWriteConfiguration(Action<SqsWriteConfigur
         }
 
         /// <inheritdoc />
-        void IPublicationBuilder<T>.Configure(JustSayingFluently bus)
+        void IPublicationBuilder<T>.Configure(JustSayingBus bus, IAwsClientFactoryProxy proxy, ILoggerFactory loggerFactory)
         {
-            bus.WithSqsMessagePublisher<T>(ConfigureWrites);
+            var logger = loggerFactory.CreateLogger<QueuePublicationBuilder<T>>();
+
+            logger.LogInformation("Adding SQS publisher for message type '{MessageType}'.",
+                typeof(T));
+
+            var config = bus.Config;
+
+            var writeConfiguration = new SqsWriteConfiguration();
+            ConfigureWrites?.Invoke(writeConfiguration);
+            writeConfiguration.ApplyQueueNamingConvention<T>(config.QueueNamingConvention);
+
+            foreach (var region in config.Regions)
+            {
+                var regionEndpoint = RegionEndpoint.GetBySystemName(region);
+                var sqsClient = proxy.GetAwsClientFactory().GetSqsClient(regionEndpoint);
+
+                var eventPublisher = new SqsPublisher(
+                    regionEndpoint,
+                    writeConfiguration.QueueName,
+                    sqsClient,
+                    writeConfiguration.RetryCountBeforeSendingToErrorQueue,
+                    bus.SerializationRegister,
+                    loggerFactory)
+                {
+                    MessageResponseLogger = config.MessageResponseLogger
+                };
+
+                if (!eventPublisher.ExistsAsync().GetAwaiter().GetResult())
+                {
+                    eventPublisher.CreateAsync(writeConfiguration).GetAwaiter().GetResult();
```

Comment:
Is this the bit you were referring to @slang25?

Reply:
@slang25 PR'd into this to resolve all of these - we now only call `GetAwaiter().GetResult()` in the DI wireup, so non-DI scenarios could now be all-async. I think it'd be good to consider how we can pull this out of DI after this PR 🙂